one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EntitySearchController.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2012-2020 MarkLogic Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.marklogic.hub.oneui.controllers;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.dataservices.EntitySearchService;
+import com.marklogic.hub.oneui.managers.EntitySearchManager;
+import com.marklogic.hub.oneui.models.Document;
+import com.marklogic.hub.oneui.models.HubConfigSession;
+import com.marklogic.hub.oneui.models.SearchQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Optional;
+
+
+@Controller
+@RequestMapping(value = "/api/entitySearch")
+public class EntitySearchController {
+
+    @Autowired
+    private HubConfigSession hubConfig;
+
+    @Autowired
+    private EntitySearchManager entitySearchManager;
```

Comment:
Shouldn't need this, since you have the method below this to construct an EntitySearchManager.